CONTRIBUTING.md

```diff
@@ -27,9 +27,14 @@ If you are reporting a security bug or vulnerability, see SECURITY.md.
 
 ## Developer Certificate of Origin
 
-We respect the intellectual property rights of others and we want to make sure all incoming contributions are correctly attributed and licensed. A Developer Certificate of Origin (DCO) is a lightweight mechanism to do that. The DCO is a declaration attached to every commit. In the commit message of the contribution, the developer simply adds a `Signed-off-by` statement and thereby agrees to the DCO, which you can find below or at [DeveloperCertificate.org](http://developercertificate.org/).
+We respect the intellectual property rights of others and we want to make sure
```

Comment:
I got kind of stuck on this. Maybe we split it by sentence instead of by some weird line number? That is how redis-doc used to work, I don't really like these artificial line breaks (since they aren't necessary).

Reply:
The line was very long. It is hard to read even on a large screen. Also the line breaks don't have effect on the final rendering. 

Are you suggesting something like the below? using an empty line to emulate the line break. Do you suggest breaking on the complete sentence only or it is ok to break after punctuations? 

```
We respect the intellectual property rights of others and we want to make sure all incoming contributions are correctly attributed and licensed. 

A Developer Certificate of Origin (DCO) is a lightweight mechanism to do that. 

The DCO is a declaration attached to every commit. In the commit message of the contribution, the developer simply adds a `Signed-off-by` statement and thereby agrees to the DCO, which you can find below or at [DeveloperCertificate.org](http://developercertificate.org/)

```

Reply:
Yeah, as a line break after each sentence. Also encourages shorter sentences.

Reply:
I agree we should aim for short sentences but it feels unnecessary coupling to me. I also don't think this is how the old codebase/doc is done (and we know consistency wasn't a strength anyways). I feel most people would be more accustomed to a fixed (or relatively fixed) line width. Maybe now it is a good time to get alignment? What do you think about 120 characters per line for everything, doc and code? FWIW, I would be happy with anything between 100-140.

Reply:
I agree generally at 120 character lines, but I'm not sure it applies to markdown files. Redis-doc was the counter point that was pretty consistent, and it is all markdown. The problem with 120 character lines is we don't enforce it though, we should add a linter like was suggested.

Reply:
In markdown files, a blank line makes a new paragraph. A single line break has no effect on the rendered HTML. It doesn't create a line break. It's just for source code readability. So when we don't want a new paragraph, just do a linebreak between sentences.

Reply:
> I agree generally at 120 character lines

SG. 

>I don't really like losing the history on this specific thing

I think there is a way to enforce the new style on the **new code**. I will experiment with clang-format next and see if I can set it up in a way that is automatically applied to the code changes only during the merge. Will tackle it in separate PRs.

> Valkey-doc is the counter point that was pretty consistent

I think this is the beauty of markdown and you can write however you like but still render perfectly. The problem I see is for someone who needs to work on the **raw** content, it becomes challenging to deal with super long lines. In this specific case, the styling changes were done on the DCO section, which is added right before this PR so I think the history part is ok. Also I feel that the doc history is less critical as the code history, in general. 

Lastly, I think the history preserving problem should be solved by the proper tooling. The diffing tool needs to be smart enough to understand the language it deals with and ideally has an option to show **semantics** differences only. I will do some research in the background next.